line_profiler/line_profiler.py

```diff
@@ -150,6 +167,25 @@ def add_module(self, mod):
 
         return nfuncsadded
 
+    def _get_wrapper_info(self, func):
+        info = getattr(func, self._profiler_wrapped_marker, None)
+        return info, bool(info and id(self) == info.profiler)
+
+    # Override these mixed-in bookkeeping methods to take care of
+    # potential multiple profiler sequences
+
+    def _already_a_wrapper(self, func):
+        return self._get_wrapper_info(func)[1]
+
+    def _mark_wrapper(self, wrapper):
+        # Are re-wrapping an existing wrapper (e.g. created by another
+        # profiler?)
+        wrapped = wrapper.__wrapped__
+        info = getattr(wrapped, self._profiler_wrapped_marker, None)
+        new_info = _WrapperInfo(info.func if info else wrapped, id(self))
```

Comment:
What happens when the function is already owned? It forgets that it was wrapped by a different instance of a LineProfiler and only remembers the latest? Could that cause problems?